### PR TITLE
trops cleanup q4

### PR DESCRIPTION
**Pull request by jalbrekt85**
handles #908

### CODE REVIEW

File: scripts/issue/908/trops_cleanup_q4.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    eurs = trops.contract(r.treasury_tokens.EURS)
+    usdc = trops.contract(r.treasury_tokens.USDC)
+    fei = trops.contract(r.treasury_tokens.FEI)
+    frax = trops.contract(r.treasury_tokens.FRAX)
+
+    bvecvx = trops.contract(r.treasury_tokens.bveCVX)
+    bcvxcrv = trops.contract(r.treasury_tokens.bcvxCRV)
+    gravi = trops.contract(r.sett_vaults.graviAURA)
+    crvIbBTC = trops.contract(r.treasury_tokens.crvIbBTC)
```
Comment (gosuto-inzasheru) [inline scripts/issue/908/trops_cleanup_q4.py:27]:
```suggestion
    ibbtc = trops.contract(r.treasury_tokens.ibBTC)
```

File: scripts/issue/908/trops_cleanup_q4.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    trops.take_snapshot(tokens=[bvecvx, bcvxcrv, gravi, crvIbBTC.address])
```
Comment (gosuto-inzasheru) [inline scripts/issue/908/trops_cleanup_q4.py:29]:
```suggestion
    trops.take_snapshot(tokens=[bvecvx, bcvxcrv, gravi, ibbtc])
```

File: scripts/issue/908/trops_cleanup_q4.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    tree.take_snapshot(tokens=[bcvxcrv])
+    vault.take_snapshot(tokens=[crvIbBTC.address, weth])
```
Comment (gosuto-inzasheru) [inline scripts/issue/908/trops_cleanup_q4.py:31]:
```suggestion
    vault.take_snapshot(tokens=[ibbtc, weth])
```

File: scripts/issue/908/trops_cleanup_q4.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    trops.badger.claim_all()
+
+    bcvxcrv.transfer(tree, TREE_DEFICIT)
+    crvIbBTC.transfer(vault, crvIbBTC.balanceOf(trops))
```
Comment (gosuto-inzasheru) [inline scripts/issue/908/trops_cleanup_q4.py:36]:
```suggestion
    ibbtc.transfer(vault, ibbtc.balanceOf(trops))
```

**Review by gosuto-inzasheru — COMMENTED** [review]
the crv_ibbtc is fine in trops for now, was just about cleaning up that ibbtc

**Comment by gosuto-inzasheru** [comment]
actually can you add a sweep of all graviaura to the voter msig? the strat/module will then withdraw and lock for vlaura.

**Comment by jalbrekt85** [comment]
> actually can you add a sweep of all graviaura to the voter msig? the strat/module will then withdraw and lock for vlaura.

f73d833

**Review by gosuto-inzasheru — APPROVED** [review]
yes looks good!

watch out with posting; tree claim needs to be posted and signed within the same cycle

**Review by sajanrajdev — APPROVED** [review]
- Transfers all gravi to voter
- Transfer all ibBTC and wETH to vault
- Claims all rewards from tree (including 12373.266529951520055332 bcvxCRV)
image
- Transfers bcvxCRV deficit to the tree (5,793.300941993312325355 + 12373.266529951520055332 - 7200 = 10966.5674719, as shown on snapshot)
- Market sells eurs, fei, frax for usdc with a 4hr deadline

Snapshots LG

As Gos mentioned, requires claim to happen during the same cycle of posting otherwise proof will be obsolete.

**Pull request merged**